src/ora-dashboard/data-table/DataTable.jsx

```diff
@@ -0,0 +1,76 @@
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import React, { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { useParams } from 'react-router';
+import LoadingOrValue from '../../components/value-or-loading/ValueOrLoading';
+import messages from './messages';
+import { fetchOraReports } from '../data/thunks';
+import { oraDataShape } from '../../data/constants';
+
+function DataTable({ intl, data }) {
+  const { courseId } = useParams();
+  const dispatch = useDispatch();
+  useEffect(() => {
+    Object.keys(data).map((blockId) => data[blockId].status || dispatch(fetchOraReports(courseId, blockId)));
```

Comment:
Just a note: this is fine here as-is for now, but each API call to the ORA reports endpoint already contains data for all ORA blocks in a course.

The way it's currently done in the LMS is somewhat hacky (use the first ORA block to retrieve data), but there's already a ticket implement and use proper API endpoints for this.

This is good to keep as is for now.

Reply:
Good to know! I wanted to have something working in the boilerplate since it makes things easier to follow.